backend/business-partner-agent/src/main/java/org/hyperledger/bpa/persistence/repository/BPAProofTemplateRepository.java

```diff
@@ -17,13 +17,29 @@
  */
 package org.hyperledger.bpa.persistence.repository;
 
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.data.annotation.Query;
 import io.micronaut.data.jdbc.annotation.JdbcRepository;
+import io.micronaut.data.model.Page;
+import io.micronaut.data.model.Pageable;
 import io.micronaut.data.model.query.builder.sql.Dialect;
-import io.micronaut.data.repository.CrudRepository;
+import io.micronaut.data.repository.PageableRepository;
+import lombok.NonNull;
 import org.hyperledger.bpa.persistence.model.BPAProofTemplate;
 
 import java.util.UUID;
 
 @JdbcRepository(dialect = Dialect.POSTGRES)
-public interface BPAProofTemplateRepository extends CrudRepository<BPAProofTemplate, UUID> {
-}
+public interface BPAProofTemplateRepository extends PageableRepository<BPAProofTemplate, UUID> {
+
+    @Query(
```

Comment:
I see copy paste. You can write this shorter, for example:

`select * from bpa_proof_template where name ilike concat('%',:name,'%');`

Reply:
yes i had it like this but then an Error occurs: 
`ERROR RouteExecutor - Unexpected error occurred: SQL Error executing Query: ERROR: missing FROM-clause entry for table "bpaproof_template_"  Position: 79`

as far as i understood, the annotation methods create aliases for the tables where "bpa_proof_template" is aliased as "bpaproof_template_", and this is the reason why the queries, when sorting in the tables is used, cannot find the table referenced by the FROM-clause. So using the "copy paste"-part u mentioned, solved this issue because the FROM-clause is recognized.

Reply:
I see, but there is nothing keeping you from keeping this short:
`select * from bpa_proof_template t where t.name ilike concat('%',:name,'%');`

Reply:
when using the sort functionality of the tables, the query is extended by the param "q" for example "q = createdAt", and the extended part refers to the aliasing i posted before. Therefore, the sorting keeps me from keeping it short, sadly.

Reply:
makes sense, then leave it as it is.